xarray_eopf/source.py

```diff
@@ -23,6 +23,35 @@ def normalize_source(source: Any, storage_options: Mapping[str, Any] | None) ->
     return source
 
 
+def get_source_paths(source: Any) -> tuple[str, str] | tuple[None, None]:
+    """
+    Derive a Zarr root path (or URL) and subgroup path
+    from given `source` object.
+
+    Args:
+        source: The source object.
+
+    Returns:
+        A tuple comprising a Zarr root path and subgroup path,
+        or the tuple `(None, None)` if the paths cannot be derived.
+        The subgroup path will be the empty string if it is
+        not part of the path derived from `source`.
+    """
+    path: str | None = None
+    if isinstance(source, (str, Path)):
+        path = source
+    elif hasattr(source, "path"):
+        path = source.path
+    elif hasattr(source, "root"):
+        path = source.root
+    if isinstance(path, (str, Path)):
+        path_parts = str(path).split(".zarr/", maxsplit=1)
+        if len(path_parts) == 2:
+            return path_parts[0] + ".zarr", path_parts[1]
```

Comment:
It looks as though this won't handle an upper-case ".ZARR" suffix -- is that OK?